packages/core-components/src/layout/Header/Header.tsx

```diff
@@ -49,7 +49,6 @@ const useStyles = makeStyles<BackstageTheme>(
       zIndex: 100,
       display: 'flex',
       flexDirection: 'row',
-      flexWrap: 'wrap',
```

Comment:
Hm I wonder if this has other ill effects. I had to fiddle pretty thoroughly to get the wrapping of the header _as a whole_ working correctly at all screen sizes. I remember there was something about how the boxes on the right stacked together with the title, and as the screen shrunk, they needed to be in a pretty carefully crafted flex box to properly wrap to the next row below.

Would be good to try this on mobile screen sizes just to be sure. Shrinking it all the way down to iphone 6 sizes etc, the label boxes on the side should stay neatly stacked and have sensible sizes and so on, irrespectively of the amount of text (and absence/presence) in the title and subtitle.

Reply:
Ah true, do check mobile, may need a media query here

Reply:
@freben mobile is a good point - there we probably want wrapping, so what about adding a breaking point where it wraps?

Reply:
@freben see https://github.com/backstage/backstage/pull/9180/commits/405176d8add2619872868
<img width="609" alt="Skärmavbild 2022-01-26 kl  11 43 07" src="https://user-images.githubusercontent.com/25153114/151149329-8c0fc834-bdde-4532-86af-c85d2ed4489a.png">
04d6a9a73041113e049

Reply:
I'm honestly not sure, I'm not great at these things :) I'd appreciate suggestions. All I know is, one gets surprised at how hard it is to find a one-size-fits-all solution :) Without the wrap here, if you have 10 label boxes, I think those will all be stacked 10 on top of each other right? At least on some screen sizes. Instead of flowing for example 5 wide in 2 rows _below_ the title and subtitle.

Reply:
This is just a suggestion of change (as we have a need for it for TechDocs) so I guess it depends on how we want it to behave. Also open to make the styling overridable here by taking in classes prop or something like that if you prefer.

But having wrap set on the header will let the children wrap into multiple lines (if its not set - it will force them to stay on one line). This means, when the subtitle takes up the entire width - the container of headerlabels will wrap and end up at a new line. As you can see example of in the screenshots above. But it does not say anything about how one specific headerlabel will wrap, that can be set on the headerlabel container if we want. You can see example of many headerlabel boxes above too, those will be stacked on top of each others (next to the title and subtitle, on the same line) and not moved down under.

Reply:
I poked around a bit in the PRs current shape, and it seems to behave decently - I say let's ship it and tweak further if any oddness arises.